elementary/clients/teams/client.py

```diff
@@ -0,0 +1,110 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from pymsteams import cardsection, connectorcard, potentialaction  # type: ignore
+from retry import retry
+
+from elementary.config.config import Config
+from elementary.tracking.tracking_interface import Tracking
+from elementary.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+OK_STATUS_CODE = 200
+
+
+class TeamsClient(ABC):
```

Comment:
In Slack, we have a webhook client and a web client, both inherit from a base SlackClient. In MSTeams we don't have the web client so I'm not sure we need this extra parent class here. WDYT?

Reply:
Yeah think you are right, just chose a similar approach as to the Slack so it would easier to also use MSTeams bots or something. But that is just an assumption that is not really necessary.

Do you prefer to simplify this?

Reply:
I've added a new [commit](https://github.com/elementary-data/elementary/pull/1397/commits/b0659fd12614b5b44a988adc0cc669cc20bb6f2d) to simplify this.